### DISCUSSION TITLE
Get Pub Key Address from ID Address

### DISCUSSION THREAD

**Discussion by dsbrgg** [filecoin-project/lotus#9118] [General Q&A] [answered]
Hello!

I've checked [this specs](https://spec.filecoin.io/appendix/address/) and noticed the following:
- There is no map from ID-address to pubkey address.

Even though this information states there's no map for both, I wonder if there's an RPC call that could resolve somehow the ID Address to Pub Key address or if they're really independent.

Also, if that's not possible, could the inverse be done (eg. Pub Key -> ID)?

**Comment by jennijuju** [comment, marked as answer]
https://github.com/filecoin-project/lotus/blob/ab0592231ca2bc71fc97673565504b852c2a6b95/api/api_full.go#L529-L531

**Reply by geoff-vball** [reply]
Also `StateLookupRobustAddress()` for non-account addresses (miners, multisigs)
